pkg/platform/runtime/artifact/buildPlan.go

```diff
@@ -0,0 +1,215 @@
+package artifact
+
+import (
+	"fmt"
+
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/multilog"
+	model "github.com/ActiveState/cli/pkg/platform/api/graphql/model/buildplanner"
+	monomodel "github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/go-openapi/strfmt"
+	"github.com/thoas/go-funk"
+)
+
+// ArtifactBuildPlan comprises useful information about an artifact that we extracted from a build plan
+type ArtifactBuildPlan struct {
+	ArtifactID       ArtifactID
+	Name             string
+	Namespace        string
+	Version          *string
+	RequestedByOrder bool
+
+	GeneratedBy strfmt.UUID
+
+	Dependencies []ArtifactID
+}
+
+// ArtifactBuildPlanMap maps artifact ids to artifact information extracted from a build plan
+type ArtifactBuildPlanMap map[ArtifactID]ArtifactBuildPlan
+
+// ArtifactNamedBuildPlanMap maps artifact names to artifact information extracted from a build plan
+type ArtifactNamedBuildPlanMap map[string]ArtifactBuildPlan
+
+// NameWithVersion returns a string <name>@<version> if artifact has a version specified, otherwise it returns just the name
+func (a ArtifactBuildPlan) NameWithVersion() string {
+	version := ""
+	if a.Version != nil {
+		version = fmt.Sprintf("@%s", *a.Version)
+	}
+	return a.Name + version
+}
+
+func NewMapFromBuildPlan(build *model.Build) ArtifactBuildPlanMap {
```

Comment:
Suggest making this take the `build.Artifacts`, `build.Steps`, etc. as separate input params so we don't need to tie it so directly to the buildplan types.

Reply:
`Steps` and `Sources` are already tied to the build plan. `Artifacts` not so much but I feel like this change is unnecessary and would also bloat the function signature.